Plugins/Area/Area.cpp

```diff
@@ -723,4 +724,31 @@ ArgumentStack Area::SetTileAnimationLoop(ArgumentStack&& args)
     return stack;
 }
 
+ArgumentStack Area::TestDirectLine(ArgumentStack&& args)
+{
+    ArgumentStack stack;
+
+    if (auto *pArea = area(args))
+    {
+        const auto fStartX = Services::Events::ExtractArgument<float>(args);
+          ASSERT_OR_THROW(fStartX >= 0.0f);
+        const auto fStartY = Services::Events::ExtractArgument<float>(args);
+          ASSERT_OR_THROW(fStartY >= 0.0f);
+        const auto fEndX = Services::Events::ExtractArgument<float>(args);
+          ASSERT_OR_THROW(fEndX >= 0.0f);
+        const auto fEndY = Services::Events::ExtractArgument<float>(args);
+          ASSERT_OR_THROW(fEndY >= 0.0f);
+        const auto fPerSpace = Services::Events::ExtractArgument<float>(args);
+          ASSERT_OR_THROW(fPerSpace >= 0.0f);
+        const auto fHeight = Services::Events::ExtractArgument<float>(args);
+            ASSERT_OR_THROW(fHeight >= 0.0f);
+        const auto bIgnoreDoors = Services::Events::ExtractArgument<int32_t>(args);
```

Comment:
I'd do `const auto bIgnoreDoors = Services::Events::ExtractArgument<int32_t>(args) != 0;` here so it's always 1 or 0.